### ISSUE TITLE
AI units ignore terrain movement cost.

### ISSUE THREAD

**Issue by calloatti** [LoneGazebo/Community-Patch-DLL#10766]
_1. Mod Version (X.Y.Z). Current Version: 4.5.0_

4.5.1

_2. DirectX Version_

_3. List of Other Mods_

IGE

_4. Describe the Issue_

I have noticed many times that AI units seem to totally ignore terrain movement cost, for example an archer moving one hex in desert and then shooting, or in this case, a Tercio that crosses a river and then attacks my cannon.

Is this intended? If it is, it seems pretty unfair to me.

Before:
![image](https://github.com/LoneGazebo/Community-Patch-DLL/assets/12817862/78a3b2e0-1b64-4253-a1da-7267b686a1b2)

After: Tercio crosses river, and then attacks, all in one turn:
![image](https://github.com/LoneGazebo/Community-Patch-DLL/assets/12817862/426e5858-1f75-4a90-96d0-c0d7991ace8d)

______________________________________________________________________________________
_5. Save Game From 1 Turn Before (ALWAYS ATTACH THIS IF POSSIBLE)_

_6. Logs (ALWAYS ATTACH THESE IF POSSIBLE)_

_7. CvMiniDump.dmp File (ATTACH IF REPORTING A GAME CRASH)_

_8. Steps to reproduce the Issue (Optional)_

_9. Screenshots of the Issue (Optional)_

**Comment by ilteroi** [comment]
well depends on promotions ... maybe the tercio is amphibious?

**Comment by calloatti** [comment]
Could be, but in 1394 AD? Also does not explain archers moving one hex in desert and shooting. I just posted what I have observed, no idea if this is easy to check or not.

**Comment by ilteroi** [comment]
have you tried the promotionflags mod

**Comment by axatin** [comment]
Barbarians sometimes have promotions that gives them double moves in deserts

**Comment by calloatti** [comment]
> have you tried the promotionflags mod

Now that you mention it, I dont see the promotion flags in units. I thought this had been integrated into the mod? At least I have the setting enabled in Interface options but it seems it does nothing?

![image](https://github.com/LoneGazebo/Community-Patch-DLL/assets/12817862/09d84c06-9653-49a5-a257-d58c323f7058)

**Comment by azum4roll** [comment]
Reload the save for it to take effect.

**Comment by azum4roll** [comment]
Please start a new issue if you can reproduce and upload a save file.
